cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTable.java

```diff
@@ -590,6 +597,24 @@ private void upsertInternal(Collection<Field<?>> fields) throws IOException {
     }
   }
 
+  private void updateRangeInternal(Range keyRange, Collection<Field<?>> fields) throws IOException {
+    String sqlQuery = getUpdateRangeSqlQuery(keyRange, fields);
+    try (PreparedStatement statement = connection.prepareStatement(sqlQuery)) {
+      int index = 1;
+      // set field values to update
+      for (Field<?> field : fields) {
+        setField(statement, field, index);
+        index++;
+      }
+      setStatementFieldByRange(keyRange, statement);
```

Comment:
I think it should be `setStatementFieldByRange(keyRange, statement, index);` Since we've set values for the fields to update, we should skip to the range part